libs/model/src/token/CreateToken.command.ts

```diff
@@ -22,50 +79,33 @@ export function CreateToken(): Command<
   return {
     ...schemas.CreateToken,
     auth: [isAuthorized({ roles: ['admin'] })],
-    body: async ({ actor, payload }) => {
-      const {
-        base,
-        chain_node_id,
-        name,
-        symbol,
-        description,
-        icon_url,
-        community_id,
-        launchpad_contract_address,
-      } = payload;
-
-      const token = await models.Token.findOne({
-        where: { name },
-      });
-      if (token) throw new InvalidInput(CreateTokenErrors.TokenNameExists);
-
-      const baseCommunity = await models.Community.findOne({
-        where: { base, id: community_id },
-      });
-      mustExist('Community Chain Base', baseCommunity);
+    body: async ({ payload }) => {
+      const { chain_node_id, transaction_hash, description, icon_url } =
+        payload;
 
-      const node = await models.ChainNode.findOne({
+      const chainNode = await models.ChainNode.findOne({
         where: { id: chain_node_id },
+        attributes: ['eth_chain_id', 'url', 'private_url'],
       });
-      mustExist(`Chain Node`, node);
 
-      if (base === ChainBase.Ethereum && !node.eth_chain_id)
-        throw new InvalidInput(CreateTokenErrors.InvalidEthereumChainId);
+      mustExist('Chain Node', chainNode);
 
-      const createdToken = await models.Token.create({
-        base,
-        chain_node_id,
-        name,
-        symbol,
-        description,
-        icon_url,
-        author_address: actor.address || '',
-        community_id,
-        launchpad_contract_address,
-        // uniswap_pool_address, - TODO: add when uniswap integration is done
-      });
+      const web3 = new Web3(chainNode.private_url || chainNode.url);
 
-      return createdToken!.toJSON();
+      const txReceipt = await web3.eth.getTransactionReceipt(transaction_hash);
+      const tokenAddress = txReceipt.logs[0].address!;
+      if (!tokenAddress) {
+        throw Error(
+          'Failed to find tokenAddress is token creation command. Review tokenAddress logic',
+        );
+      }
```

Comment:
again, this is just for parsing the logs for this specific transaction type (tokenLaunch) to obtain the tokenAddress. I don't think it makes sense throwing it in a util since it will only be used here.